app/Models/Server.php

```diff
@@ -453,6 +455,49 @@ public function resources(): array
         });
     }
 
+    public function formatResource(string $resourceKey, bool $percentage = false, bool $limit = false, bool $time = false): string
+    {
+        $resourceAmount = $this->{$resourceKey} ?? 0;
+
+        if (!$limit) {
+            $resourceAmount = $this->resources()[$resourceKey] ?? 0;
+        }
+
+        if ($time) {
+            return self::timeFormatter($resourceAmount);
+        }
+
+        return self::byteFormatter($resourceAmount, percentage: $percentage, limit: $limit);
+    }
+
+    public static function timeFormatter(int $amount): string
+    {
+        if ($amount === 0) {
+            return 'Offline';
+        }
+
+        return now()->subMillis($amount)->diffForHumans(syntax: CarbonInterface::DIFF_ABSOLUTE, short: true, parts: 2);
+    }
+
+    public static function byteFormatter(int $amount, int $precision = 3, bool $percentage = false, bool $limit = false): string
+    {
+        $bits = config('panel.use_binary_prefix') ? 1024 : 1000;
+        $unitLetters = ['', 'K', 'M', 'G', 'T', 'P', 'E', 'Z', 'Y', 'B'];
+        $unitLetter = $unitLetters[$precision];
+        $unitSuffix = config('panel.use_binary_suffix') ? 'i' : '';
+        $unit = ' ' . $unitLetter . $unitSuffix . 'B';
+
+        if ($amount === 0 & $limit) {
+            return 'Unlimited';
+        }
+
+        if ($percentage) {
+            $unit = '%';
+        }
+
+        return Number::format($amount / $bits ** $precision, maxPrecision: 2, locale: auth()->user()->language) . $unit;
+    }
```

Comment:
There is already a [`convert_bytes_to_readable` helper function](https://github.com/pelican-dev/panel/blob/main/app/helpers.php#L22) btw.

Also, this seems to be unhandy/ confusing for percentage values.

Reply:
Thanks, I've reused that method!